### PR TITLE
Externalise backingstore and bucketclass errors - core side

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. Added tiering info mode and calculation

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1.

### CODE REVIEW

File: src/api/tiering_policy_api.js [hunk 1]
```diff
@@ -217,5 +218,23 @@ module.exports = {
                 'OPTIMAL'
             ]
         },
+        tiering_policy_mode: {
```
Comment (nimrod-becker) [inline src/api/tiering_policy_api.js]: Is it the same as bucket_api::bucket_mode? 
Should we have a common type for both?

File: src/server/system_services/tier_server.js [hunk 2]
```diff
@@ -680,6 +680,37 @@ function get_tiering_policy_info(tiering_policy, tiering_pools_status,
     return info;
 }
 
+function calc_tiering_policy_mode(tiers, metrics) {
```
Comment (nimrod-becker) [inline src/server/system_services/tier_server.js]: What's the relation to calc_bucket_mode?
Should we somehow use one another?

File: src/server/system_services/bucket_server.js [hunk 3]
```diff
@@ -1411,7 +1411,9 @@ function get_bucket_info({
     // calc_bucket_aggregated_mode(metrics);
     info.mode = bucket.namespace ?
         calc_namespace_mode() :
-        calc_bucket_mode(tiering.tiers, metrics);
+        calc_bucket_mode(tiering.tiers, metrics, false);
```
Comment (nimrod-becker) [inline src/server/system_services/bucket_server.js]:
```suggestion
        calc_bucket_mode(tiering.tiers, metrics, false /*ignore_quota*/);
```

File: src/server/system_services/bucket_server.js [hunk 4]
```diff
@@ -1411,7 +1411,9 @@ function get_bucket_info({
...
+
+    info.tiering.mode = calc_bucket_mode(tiering.tiers, metrics, true);
```
Comment (nimrod-becker) [inline src/server/system_services/bucket_server.js]:
```suggestion
    info.tiering.mode = calc_bucket_mode(tiering.tiers, metrics, true /*ignore_quota*/);
```

**Review by dannyzaken — APPROVED** [review]

**Pull request merged**